File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        Map<Integer, Integer> freq = new HashMap<>();
        freq.put(0, 1);
        
        int count = 0;
        int runningSum = 0;
        for (int num : nums) {
            runningSum += num;
            count += freq.getOrDefault(runningSum - k, 0);
            freq.put(runningSum, freq.getOrDefault(runningSum, 0) + 1);
            
        }
        
        return count;
    }
}

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        count = 0
        currsum = 0
        length = len(nums)

        hashmap = {0: 1}
        
        for a in nums:
            currsum += a
            count += hashmap.get(currsum - k, 0)
            hashmap[currsum] = hashmap.get(currsum, 0) + 1

        return count